### ISSUE TITLE
Notebooks: cells do not have aria labels set

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#107582]
1. Turn on your screen reader
2. Navigate between cells
3. Notice how when you navigate on a non-code cell Screen Reader does not read anything

Inspecting the DOM it seems like the `aria-lable` is not set on non-code cells.
List supports this out of the box via accessibility providers so I think this should be easy to fix

reported by @webczat

**Comment by webczat** [comment]
actually sometimes it does read things. for example in case of markdown cells sometimes it probably reads the whole output, sometimes it says that it is markdown editor... hmm not sure if I can trigger different behaviors in some specific way or not

**Comment by roblourens** [comment]
What should it read, the markdown source as text?

**Comment by webczat** [comment]
currently unsure, not much experience with notebooks in any form. Not sure what identifies the cell for a sighted person? is that only the ordering and type of cell?

**Comment by roblourens** [comment]
Yes, that's it. There's no label or anything useful like that. Checking other notebook systems, Jupyter doesn't read anything when the cell is selected. When focus goes into a cell editor, it starts reading the full document content (starting from the top instead of the selected cell). Google Colab reads the names/keyboard shortcuts of all commands in the cell toolbar when a cell is selected, followed by the cell contents. So I don't think there's a good precedent and we'll have to make it up as we go.

**Comment by webczat** [comment]
Well, I did have an idea, I mean, sometimes vscode would read just the type of editor like markdown, I don't remember how it reads that exactly, probably something like "markdown editor", and it seems more appropriate than something like reading it top to bottom when I focus a closed markdown cell.
As for jupyter, it is not really accessible in general, unless it changed.

**Comment by rebornix** [comment]
I like the idea of reading the type. For example, "focus on a markdown cell", "focus on a code cell" and then users would at least know what would happen when pressing "Enter" (editing markdown, or focus into the code cell editor) or "Ctrl+Enter" (run the code if it's a code cell).

**Comment by webczat** [comment]
it could even go and read the beginning of the contents, but it is the least important piece of info there

**Comment by rebornix** [comment]
We now have aria support for markup cells.
